### PR TITLE
Bump flake8 to fix the AttributeError: 'QuoteChecker' on py3.8

### PR DESCRIPTION
**Pull request by thomasdesr**
## Why are these changes needed?

Bump `flake8(-quotes)?` to latest in order to stop it from having a sad when it is run with more than one file as input. This mostly happens when you run `./scripts/ci/format.sh` after having modified more than one file.

## Related issue number

n/a

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(
   - [x] Manually verified that things don't explode horribly on 3.8 & 3.6

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]

**Review by wuisawesome — APPROVED** [review]

**Comment by rkooo567** [comment]
There's a lint failure. Please fix it!

**Comment by wuisawesome** [comment]
Looks like bumping the lint version actually changes the formatting rules 😐. I vote we just fix this and do it anyways.

**Comment by thomasdesr** [comment]
Agreed, I'll clean those up in the next day or so and re-flag for review.

File: python/ray/tune/trial.py [hunk 1]
```diff
@@ -301,7 +301,7 @@ def __init__(self,
         if trial_dirname_creator:
             self.custom_dirname = trial_dirname_creator(self)
             if os.path.sep in self.custom_dirname:
-                raise ValueError(f"Trial dirname must not contain '/'. "
+                raise ValueError("Trial dirname must not contain '/'. "
```
Comment (wuisawesome) [inline python/ray/tune/trial.py:304]: @richardliaw btw this looks like a bug (looks like it was here before the PR)

**Review by wuisawesome — APPROVED** [review]
lgtm, mostly just lots of boilerplate fixes

thanks for going through the effort!

File: python/ray/tune/trial.py [hunk 2]
```diff
@@ -301,7 +301,7 @@ def __init__(self,
...
                                  "Got {self.custom_dirname}")
```
Comment (richardliaw) [inline python/ray/tune/trial.py:305]:
```suggestion
                                 f"Got {self.custom_dirname}")
```

Comment (richardliaw) [reply python/ray/tune/trial.py:305]: (mind i piggyback this fix)

Comment (thomasdesr) [reply python/ray/tune/trial.py:305]: Good catch! I fixed another one of these, thanks for catching this one :D

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Pull request closed without merging**